streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -999,7 +1006,9 @@ public synchronized Pattern sourceTopicPattern() {
         return this.topicPattern;
     }
 
-    public synchronized void updateSubscriptions(SubscriptionUpdates subscriptionUpdates) {
+    public synchronized void updateSubscriptions(SubscriptionUpdates subscriptionUpdates, String threadId) {
```

Comment:
Feels a little kludgy but since `TopologyBuilder` is shared among threads, thought this was best approach